src/core/mac/mac.cpp

```diff
@@ -257,6 +257,11 @@ bool Mac::IsEnergyScanInProgress(void)
     return (mState == kStateEnergyScan) || (mPendingScanRequest == kScanTypeEnergy);
 }
 
+bool Mac::IsInTransmitState(void)
+{
+    return (mState == kStateTransmitData) || (mState == kStateTransmitBeacon);
```

Comment:
That's right. It seems that it is not possible right now to check if MAC is transmitting Beacon Request. To do that we should add a new state to the mState enum or additional field to the structure. Perhaps we could just state in the documentation that IsInTranmitState() does not cover Beacon Requests right now. I check for active scan, energy scan and transmit states separately in the application. Or should I add this additional field to indicate transmission of all command frames in IsInTransmitState?